src/Bicep.Core/TypeSystem/TypeValidator.cs

```diff
@@ -301,7 +303,20 @@ private static TypeSymbol NarrowObjectType(ITypeManager typeManager, ObjectSynta
 
             if (missingRequiredProperties.Any())
             {
-                diagnostics.Add(DiagnosticBuilder.ForPosition(expression).MissingRequiredProperties(ShouldWarn(targetType), missingRequiredProperties));
+                TextSpan span = expression.Span;
+                var parent = typeManager.GetParent(expression);
+
+                if (parent is ObjectPropertySyntax p1)

```

Comment:
Can you give `p1` & `p2` more descriptive names please?
What about `objectPropertyParent` and `declarationParent`?

Reply:
done